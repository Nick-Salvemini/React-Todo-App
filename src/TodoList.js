import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm"
import Todo from "./Todo"

function TodoList() {
    const [todos, setTodos] = useState([])

    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, { ...newTodo, id: Date.now() }])
    }

    const removeTodo = (id) => {
        setTodos(todos.filter(todo => todo.id != id));
    }

    return (
        <>
            <h1>Todo List</h1>
            <NewTodoForm addTodo={addTodo} />
            <div>
                {todos.map(todo =>
                    <Todo
                        key={todo.id}
                        task={todo.task}
                        remove={() => removeTodo(todo.id)}
                    />)}
            </div>
        </>
    );
}

export default TodoList