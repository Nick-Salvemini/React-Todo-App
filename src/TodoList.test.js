import React from "react";
import { render } from "@testing-library/react";
import TodoList from "./TodoList";

describe('TodoList component', () => {
    it('renders without crashing', () => {
        render(<TodoList />);
    });

    it('matches snapshot', () => {
        const { container } = render(<TodoList />);
        expect(container).toMatchSnapshot();
    });
});