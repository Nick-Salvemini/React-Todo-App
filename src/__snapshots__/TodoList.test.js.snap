// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component matches snapshot 1`] = `
<div>
  <h1>
    Todo List
  </h1>
  <form>
    <label
      for="task"
    >
      Task: 
    </label>
    <input
      id="task"
      name="task"
      placeholder="New Task"
      type="text"
      value=""
    />
    <button>
      Add New Task
    </button>
  </form>
  <div />
</div>
`;
